### ISSUE TITLE
Zero gradient from clipping critic gradient?

### ISSUE THREAD

**Issue by danijar** [openai/baselines#131]
The actor loss is defined as ([Line 164](https://github.com/openai/baselines/blob/699919f1cf2527b184f4445a3758a773f333a1ba/baselines/ddpg/ddpg.py#L164)):

```python
self.actor_loss = -tf.reduce_mean(self.critic_with_actor_tf)
```

Doesn't clipping `self.critic_with_actor_tf` set large critic gradients to 0 instead of to 1 ([Line 125](https://github.com/openai/baselines/blob/699919f1cf2527b184f4445a3758a773f333a1ba/baselines/ddpg/ddpg.py#L125))?

```python
self.critic_with_actor_tf = denormalize(tf.clip_by_value(self.normalized_critic_with_actor_tf, self.return_range[0], self.return_range[1]), self.ret_rms)
```

**Comment by brendenpetersen** [comment]
Line 125 isn't clipping any gradients. It's just clipping the critic's output by the environment's min/max return. This prevents the critic from predicting a return outside the possible return range of the environment.

**Comment by danijar** [comment]
What is the gradient when the value is too large and gets clipped?

**Comment by brendenpetersen** [comment]
I don't know, but it doesn't sound like an issue of clipping gradients. Clipping the critic output may be problematic for computing gradients, but not because of gradient clipping.

**Comment by danijar** [comment]
Sorry, I had a typo in the title of the issue. What I wanted to point out is that clipping the critic output and using the result in a loss function might cause gradients to be zero.
